File: app/src/main/java/com/anokmik/tripassistant/author/profile/ProfileFragment.java
package com.anokmik.tripassistant.author.profile;

import android.os.Bundle;

import com.anokmik.tripassistant.R;
import com.anokmik.tripassistant.base.BaseFragment;
import com.anokmik.tripassistant.databinding.FragmentProfileBinding;

public final class ProfileFragment extends BaseFragment<FragmentProfileBinding> implements ProfileContract.View {

    private static final String KEY_AUTHOR_ID = "key_author_id";

    public static ProfileFragment newInstance(long authorId) {
        Bundle args = new Bundle();
        args.putLong(KEY_AUTHOR_ID, authorId);
        ProfileFragment fragment = new ProfileFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    protected boolean displayHomeAsUp() {
        return true;
    }

    @Override
    protected int getLayoutId() {
        return R.layout.fragment_profile;
    }

    @Override
    protected void initBinding(FragmentProfileBinding binding) {
        binding.setPresenter(new ProfilePresenter(this, getAuthorId()));
    }

    @Override
    public void setProfileTitle() {
        setActionBarTitle(R.string.title_profile);
    }

    @Override
    public void setEditAuthorTitle() {
        setActionBarTitle(R.string.title_edit_author);
    }

    @Override
    public void back() {
        getActivity().onBackPressed();
    }

    private long getAuthorId() {
        return getArguments().getLong(KEY_AUTHOR_ID);
    }

}

File: app/src/main/kotlin/com/anokmik/tripassistant/author/profile/ProfileFragment.kt
package com.anokmik.tripassistant.author.profile

import android.os.Bundle
import com.anokmik.tripassistant.R
import com.anokmik.tripassistant.base.BaseFragment
import com.anokmik.tripassistant.databinding.FragmentProfileBinding

class ProfileFragment : BaseFragment<FragmentProfileBinding>(), ProfileContract.View {

    override val displayHomeAsUp = true

    override val layoutId = R.layout.fragment_profile

    private val authorId by lazy { arguments.getLong(KEY_AUTHOR_ID) }

    override fun initBinding(binding: FragmentProfileBinding) {
        binding.presenter = ProfilePresenter(this, authorId)
    }

    override fun setProfileTitle() {
        setActionBarTitle(R.string.title_profile)
    }

    override fun setEditAuthorTitle() {
        setActionBarTitle(R.string.title_edit_author)
    }

    override fun back() {
        activity.onBackPressed()
    }

    companion object {

        private val KEY_AUTHOR_ID = "key_author_id"

        fun newInstance(authorId: Long): ProfileFragment {
            val args = Bundle()
            args.putLong(KEY_AUTHOR_ID, authorId)
            val fragment = ProfileFragment()
            fragment.arguments = args
            return fragment
        }

    }

}